sdk/src/mcm_dp.c

```diff
@@ -157,143 +179,172 @@ static void parse_memif_param(mcm_conn_param* request, memif_socket_args_t* memi
 }
 
 /* Connect to MCM media proxy daemon through memif. */
-mcm_conn_context* mcm_create_connection_proxy(mcm_conn_param* param)
+int mcm_create_connection_proxy(MeshClient mc, MeshConnection conn)
 {
     int ret = 0;
-    mcm_conn_context* conn_ctx = NULL;
-    mcm_dp_addr media_proxy_addr = {};
     int sockfd = 0;
     uint32_t session_id = 0;
     memif_conn_param memif_param = {};
 
+    MeshClientConfig *mc_conf = (MeshClientConfig*) mc;
+    MeshConnectionConfig *conn_conf= (MeshConnectionConfig*) conn;
+
+    mcm_conn_param param;
+    param.type = conn_conf->type;
+    param.protocol = conn_conf->proto;
+    param.local_addr = conn_conf->local_addr;
+    param.remote_addr = conn_conf->remote_addr;
+    param.memif_interface = conn_conf->memif_interface;
+    param.payload_type = conn_conf->payload_type;
+    param.payload_codec = conn_conf->payload_codec;
+    param.payload_args.video_args = conn_conf->payload_args.video_args;
+    param.payload_args.audio_args = conn_conf->payload_args.audio_args;
+    param.payload_args.anc_args = conn_conf->payload_args.anc_args;
+    param.width = conn_conf->width;
+    param.height = conn_conf->height;
+    param.fps = conn_conf->fps;
+    param.pix_fmt = conn_conf->pix_fmt;
+    param.payload_type_nr = conn_conf->payload_type_nr;
+    param.payload_mtl_flags_mask = conn_conf->payload_mtl_flags_mask;
+    param.payload_mtl_pacing = conn_conf->payload_mtl_pacing;
```

Comment:
This must be reworked entirely. It's so strange to see so many fields with same names are assigned one by one. Maybe use `memcpy`?